driver-core/src/main/com/mongodb/client/model/SearchIndexType.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.model;
+
+import com.mongodb.annotations.Sealed;
+import org.bson.BsonString;
+import org.bson.BsonValue;
+
+import static com.mongodb.assertions.Assertions.notNull;
+
+/**
+ * This interface represents an Atlas Search Index type, which is utilized for creating specific types of indexes.
+ * <p>
+ * It provides methods for creating and converting Atlas Search Index types to {@link BsonValue}.
+ * </p>
+ *
+ * @mongodb.server.release 7.0
```

Comment:
The `vectorSearch` type was backported to 7.0 as noted in [SERVER-83107](https://jira.mongodb.org/browse/SERVER-83107), and I noticed it has also been backported to 6.0. Consequently, I updated our Javadoc to mention 6.0 as the minimum version since we also run our tests against this version.

Additionally, I just noticed that the managing of `SearchIndexes` is documented for 6.0 in the [MongoDB documentation](https://www.mongodb.com/docs/v6.0/reference/command/createSearchIndexes/). Therefore, I have revised our Javadoc for existing search index helpers to reflect this. Please let me know if there are any concerns or objections.